src/components/survey/reusables.tsx

```diff
@@ -0,0 +1,462 @@
+    }
+    
+    setReusableDescriptionDialogOpen(true); // TODO: Find another way to set description dialog open
+  };
+
+  /**
+   * Event handler for delete survey reusable confirm
+   */
+  const onDeleteSurveyReusableConfirm = async () => {
+    if (!keycloak?.token || !selectedReusableIds || !surveyId) {
+      return;
+    }
+
+    const reusablesApi = Api.getSurveyReusablesApi(keycloak.token);
+
+    await Promise.all(
+      selectedReusableIds.map(async materialId => {
+        await reusablesApi.deleteSurveyReusable({
+          surveyId: surveyId,
+          reusableId: materialId.toString()
+        });
+      })
+    );
+
+    fetchSurveyReusables();
+    setSelectedReusableIds([]);
+    setDeletingMaterial(false);
+  };
+
+  /**
+   * Check if viewport is mobile size
+   */
+  const isMobile = useMediaQuery(theme.breakpoints.down("lg"));
+
+  /**
+   * Renders delete material dialog
+   */
+  const renderDeleteSurveyMaterialDialog = () => (
+    <GenericDialog
+      error={ false }
+      open={ deletingMaterial }
+      onClose={ () => setDeletingMaterial(false) }
+      onCancel={ () => setDeletingMaterial(false) }
+      onConfirm={ onDeleteSurveyReusableConfirm }
+      title={ strings.survey.reusables.deleteReusableDialog.title }
+      positiveButtonText={ strings.generic.confirm }
+      cancelButtonText={ strings.generic.cancel }
+    >
+      <Typography>
+        { strings.survey.reusables.deleteReusableDialog.text }
+      </Typography>
+    </GenericDialog>
+  );
+
+  /**
+   * Renders add survey reusable dialog
+   */
+  const renderAddSurveyReusableDialog = () => {
+    const reusableOptions = Object.values(reusableMaterials).map(material =>
+      <MenuItem key={ material.id } value={ material.id }>
+        { material.name }
+      </MenuItem>
+    );
+
+    const usabilityOptions = Object.values(Usability).map(usability =>
+      <MenuItem key={ usability } value={ usability }>
+        { LocalizationUtils.getLocalizedUsability(usability) }
+      </MenuItem>
+    );
+
+    const unitOptions = Object.values(Unit).map(unit =>
+      <MenuItem key={ unit } value={ unit }>
+        { LocalizationUtils.getLocalizedUnits(unit) }
+      </MenuItem>
+    );
+
+    return (
+      <GenericDialog
+        error={ false }
+        open={ addingSurveyReusable }
+        onClose={ () => setAddingSurveyReusable(false) }
+        onCancel={ () => setAddingSurveyReusable(false) }
+        onConfirm={ onAddReusableConfirm }
+        title={ strings.survey.reusables.addNewBuildinPartsDialog.title }
+        positiveButtonText={ strings.generic.confirm }
+        cancelButtonText={ strings.generic.cancel }
+      >
+        <TextField
+          fullWidth
+          color="primary"
+          label={ strings.survey.reusables.addNewBuildinPartsDialog.buildingPart }
+          onChange={ event => setNewMaterial({ ...newMaterial, componentName: event.target.value }) }
```

Comment:
Create onChange handler function that uses even target name and value to update the material